File: Beecrowd/TypeScript/beginner/beecrowd-1040.ts
import { readFileSync } from "node:fs";
import { EOL } from "node:os";
import { stdin } from "node:process";

const input = readFileSync(stdin.fd, { encoding: "utf8" }).split(EOL);
// const input = ["2.0 4.0 8.5 9.0", "6.5", ""];

const notes = input[0].split(" ").map(Number);
const exam = Number(input[1]);

const media = (notes[0] * 2 + notes[1] * 3 + notes[2] * 4 + notes[3] * 1) / 10;

console.log(`Media: ${media.toFixed(1)}`);

if (media >= 7.0) {
	console.log("Aluno aprovado.");
} else if (media < 5.0) {
	console.log("Aluno reprovado.");
} else if (media >= 5.0 && media <= 6.9) {
	console.log("Aluno em exame.");

	console.log(`Nota do exame: ${exam.toFixed(1)}`);

	const newMedia = (media + exam) / 2;

	if (newMedia >= 5.0) {
		console.log("Aluno aprovado.");
	} else {
		console.log("Aluno reprovado.");
	}
	console.log(`Media final: ${newMedia.toFixed(1)}`);
}

File: Csharp/Beecrowd/src/Resolutions/Beginner/Beecrowd1040.cs
using System.Globalization;

namespace BeeCrowd.src.Resolutions.Beginner;

public class Beecrowd1040
{
    public static void Average3()
    {
        float n1, n2, n3, n4, average, exam;

        string[] s = (Console.ReadLine() ?? string.Empty).Split(' ');

        n1 = float.Parse(s[0], CultureInfo.InvariantCulture);
        n2 = float.Parse(s[1], CultureInfo.InvariantCulture);
        n3 = float.Parse(s[2], CultureInfo.InvariantCulture);
        n4 = float.Parse(s[3], CultureInfo.InvariantCulture);

        average = ((n1 * 2 + n2 * 3 + n3 * 4 + n4 * 1) / (2 + 3 + 4 + 1));

        if (average >= 7.0)
        {
            Console.WriteLine("Media: " + average.ToString("F1", CultureInfo.InvariantCulture));
            Console.WriteLine("Aluno aprovado.");
        }
        else
        {
            if (average < 5.0)
            {
                Console.WriteLine("Media: {0:f1}", (average - 0.01).ToString("F1", CultureInfo.InvariantCulture));

                Console.WriteLine("Aluno reprovado.");
            }
            else
            {
                Console.WriteLine("Media: {0:f1}", (average - 0.01).ToString("F1", CultureInfo.InvariantCulture));
                Console.WriteLine("Aluno em exame.");
                exam = float.Parse(Console.ReadLine() ?? string.Empty, CultureInfo.InvariantCulture);
                Console.WriteLine("Nota do exame: " + exam.ToString("F1", CultureInfo.InvariantCulture));
                average = (average + exam) / 2;

                if (average >= 5.0)
                    Console.WriteLine("Aluno aprovado.");
                else
                    Console.WriteLine("Aluno reprovado.");

                Console.WriteLine("Media final: " + average.ToString("F1", CultureInfo.InvariantCulture));
            }
        }
    }
}
